ui/page/dcrdex/market.go

```diff
@@ -210,19 +207,30 @@ func NewDEXMarketPage(l *load.Load, selectServer string) *DEXMarketPage {
 // the page is displayed.
 // Part of the load.Page interface.
 func (pg *DEXMarketPage) OnNavigatedTo() {
+	if pg.isDEXReset() {
+		return
+	}
+
 	pg.ctx, pg.cancelCtx = context.WithCancel(context.Background())
 
-	noteFeed := pg.dexc.NotificationFeed()
+	dexc := pg.AssetsManager.DexClient()
+	noteFeed := dexc.NotificationFeed()
 	go func() {
 		defer func() {
 			noteFeed.ReturnFeed()
 		}()
 		for {
+			// Always check if the dex client is ready. We want to exit if there
+			// was a reset.
+			if !pg.AssetsManager.DEXCInitialized() {
```

Comment:
Is this the check we should be making earlier instead of `pg.isDEXReset()`? Or should we be using `pg.isDEXReset()` here and below?

Reply:
`pg.isDEXReset()` triggers and action that we only want to execute once but I can find a way to reuse the method here.